pike/test/copychunk.py

```diff
@@ -428,3 +453,165 @@ def test_neg_dst_is_a_dir(self):
                                                         pike.smb2.FILE_DELETE_ON_CLOSE,
                                             dst_disp=pike.smb2.FILE_OPEN_IF,
                                             exp_error=pike.ntstatus.STATUS_INVALID_DEVICE_REQUEST)
+
+    def generic_multiple_resume_key_test_case(self, sess_num):
+        """
+        multiple session resume key test
+        """
+        src_filename = "src_copy_chunk_rk.txt"
+        content = "resume key test"
+        rk_list = []
+
+        for i in range(sess_num):
+            chan, tree = self.tree_connect()
+            fh_1 = chan.create(tree,
+                               src_filename,
+                               access=access_rwd,
+                               share=share_all).result()
+            chan.write(fh_1, 0, content)
+            resume_key_1_1 = chan.resume_key(fh_1)[0][0].resume_key
+            resume_key_1_2 = chan.resume_key(fh_1)[0][0].resume_key
+            fh_2 = chan.create(tree,
+                               src_filename,
+                               access=access_rwd,
+                               share=share_all).result()
+            resume_key_2_1 = chan.resume_key(fh_2)[0][0].resume_key
+            resume_key_2_2 = chan.resume_key(fh_2)[0][0].resume_key
+            self.other_chan_list.append(
+                {"channel": chan, "tree": tree, "filehandles": [fh_1, fh_2]})
+            try:
+                # for same file handler, resume_key should be same
+                self.assertEqual(resume_key_1_1, resume_key_1_2)
+                self.assertEqual(resume_key_2_1, resume_key_2_2)
+                # for different file handler in same session
+                self.assertNotEqual(resume_key_1_1, resume_key_2_1)
+                # for different file hander and different session
+                self.assertNotIn(resume_key_1_1, rk_list)
+                self.assertNotIn(resume_key_2_1, rk_list)
+            except Exception as e:
```

Comment:
don't catch `Exception` here. and especially don't re-raise a new AssertionError since that will overwrite the actual failure message and traceback.

if you want to provide more information context, then i would suggest a construction like the following

```
except AssertionError:
    self.error("resume key check failed on session {0}".format(i), exc_info=True)
    raise
```

this will log a message providing information about where in the loop the test failed (notoriously difficult to ascertain just from a backtrace). Additionally, using the bare `raise` will preserve all information about the failure and bubble it up to higher layers (test reporting, etc).